File: typescript/tests/pages/secureExchange.ts
import { Page } from "@playwright/test"
import { InquiryForm } from "../components/inquiryForm.ts"
import { step } from "../testStep.ts"

export class SecureExchange {
  inquiryForm: InquiryForm

  constructor(private page: Page) {
    this.inquiryForm = new InquiryForm(this.page)
  }

  @step
  async goToInquiryForm() {
    await this.page.getByRole("link", { name: "Send inquiry" }).click()
  }

  @step
  async fillInquiryForm(inquiryData: {
    firstName: string
    lastName: string
    email: string
    phoneNumber: string
    message: string
  }) {
    await this.inquiryForm.fillInquiryForm(inquiryData)
  }
}

File: kotlin/src/test/kotlin/pages/SecureExchange.kt
package pages

import com.microsoft.playwright.Page
import com.microsoft.playwright.Page.GetByRoleOptions
import com.microsoft.playwright.options.AriaRole.LINK
import components.InquiryForm

class SecureExchange(private val page: Page) {

    fun `go to inquiry form`(): SecureExchange {
        page.getByRole(LINK, GetByRoleOptions().setName("Send inquiry")).click()
        return this
    }

    fun `submit inquiry form`(
        firstName: String,
        lastName: String,
        email: String,
        phoneNumber: String,
        message: String
    ): SecureExchange {
        InquiryForm(page).invoke(
            firstName = firstName,
            lastName = lastName,
            email = email,
            phoneNumber = phoneNumber,
            message = message,
        )
        return this
    }
}

